giraffe/src/components/SimpleTable/PagedTable.tsx

```diff
@@ -54,31 +56,28 @@ const getNumberOfRowsOnCurrentPage = (
   let lastSignature
   let signature
 
-  // rowHeight is now guaranteed to be greater than zero
-  const lastVisibleRowMinimumHeight = 0.2 * rowHeight
-
   while (rowIdx < result.table.length) {
     if (result.table.columns?.table?.data?.[rowIdx] !== currentTable) {
       signature = Object.values(result.table.columns)
-        .map(
-          c =>
-            `${c.name}::${c.fluxDataType}::${result.fluxGroupKeyUnion.includes(
-              c.name
-            )}`
-        )
+        .map(column => {
+          if (column.data[rowIdx] !== undefined) {
+            return `${column.name}::${
+              column.fluxDataType
+            }::${result.fluxGroupKeyUnion.includes(column.name)}`
+          }
+          return 'undefined'
+        })
```

Comment:
This is the heart of the fix. The signature, which is a string value representing a table header, did not change when a new table with a different table header should be counted. We fix this by accurately creating the signature to detect if Simple Table will create a new table header or not.